### PR TITLE
fix: python snippet and template recursion errors

### PR DESCRIPTION
**Pull request by armandobelardo**

### CODE REVIEW

File: generators/python/src/fern_python/snippet/snippet_template_factory.py [hunk 1]
```diff
@@ -360,9 +360,6 @@ def _get_single_union_type_template(
                 if template_input is not None:
                     template_inputs.append(template_input)
 
-            object_reference = self._context.pydantic_generator_context.get_class_reference_for_type_id(
```
Comment (armandobelardo) [inline generators/python/src/fern_python/snippet/snippet_template_factory.py:363]: These removals are removing dead code to limit logic

File: generators/python/src/fern_python/snippet/snippet_template_factory.py [hunk 2]
```diff
@@ -641,6 +630,10 @@ def get_type_reference_template_input(
         name_breadcrumbs: Optional[List[str]],
         indentation_level: int = 0,
     ) -> Union[TemplateInput, None]:
+        # Terminate if depth is too deep
+        if indentation_level >= self.MAXIMUM_TEMPLATE_DEPTH:
```
Comment (armandobelardo) [inline generators/python/src/fern_python/snippet/snippet_template_factory.py:634]: This is the real change, it uses indentation level to indicate depth, but happy to introduce an explicit depth param

File: generators/python/src/fern_python/generators/sdk/client_generator/endpoint_function_generator.py [hunk 3]
```diff
@@ -1298,7 +1298,7 @@ def _get_snippet_for_request_reference(
                 shape = union.shape.get_as_union()
                 if shape.type == "alias":
                     return self._get_snippet_for_request_reference(
-                        example_type_reference, is_optional, request_parameter_names
+                        shape.value, is_optional, request_parameter_names
```
Comment (armandobelardo) [inline generators/python/src/fern_python/generators/sdk/client_generator/endpoint_function_generator.py:1301]: This is the actual change for snippets

**Review by amckinney — APPROVED** [review]
lgtm - discussed exploring a higher limit offline

**Pull request merged**